src/coredata.cs

```diff
@@ -1392,6 +1452,149 @@ interface NSMigrationManager {
+
+		[Export ("changedObjectID", ArgumentSemantic.Copy)]
+		NSManagedObjectID ChangedObjectId { get; }
+
+		[Export ("changeType")]
+		NSPersistentHistoryChangeType ChangeType { get; }
+
+		[NullAllowed, Export ("tombstone", ArgumentSemantic.Copy)]
+		NSDictionary Tombstone { get; }
+
+		[NullAllowed, Export ("transaction", ArgumentSemantic.Strong)]
+		NSPersistentHistoryTransaction Transaction { get; }
+
+		[NullAllowed, Export ("updatedProperties", ArgumentSemantic.Copy)]
+		NSSet<NSPropertyDescription> UpdatedProperties { get; }
+	}
+
+	[Watch (4,0), TV (11,0), Mac (10,13), iOS (11,0)]
+	[BaseType (typeof (NSObject))]
+	interface NSPersistentHistoryToken : NSCopying, NSSecureCoding
+	{
+	}
+
+	[Watch (4,0), TV (11,0), Mac (10,13), iOS (11,0)]
+	[BaseType (typeof(NSPersistentStoreRequest))]
+	interface NSPersistentHistoryChangeRequest
+	{
+		[Static]
+		[Export ("fetchHistoryAfterDate:")]
+		NSPersistentHistoryChangeRequest FetchHistoryAfterDate (NSDate date);
+
+		[Static]
+		[Export ("fetchHistoryAfterToken:")]
+		NSPersistentHistoryChangeRequest FetchHistoryAfterToken ([NullAllowed] NSPersistentHistoryToken token);
+
+		[Static]
+		[Export ("fetchHistoryAfterTransaction:")]
+		NSPersistentHistoryChangeRequest FetchHistoryAfterTransaction ([NullAllowed] NSPersistentHistoryTransaction transaction);
+
+		[Static]
+		[Export ("deleteHistoryBeforeDate:")]
+		NSPersistentHistoryChangeRequest DeleteHistoryBeforeDate (NSDate date);
+
+		[Static]
+		[Export ("deleteHistoryBeforeToken:")]
+		NSPersistentHistoryChangeRequest DeleteHistoryBeforeToken ([NullAllowed] NSPersistentHistoryToken token);
+
+		[Static]
+		[Export ("deleteHistoryBeforeTransaction:")]
+		NSPersistentHistoryChangeRequest DeleteHistoryBeforeTransaction ([NullAllowed] NSPersistentHistoryTransaction transaction);
+
+		[Export ("resultType", ArgumentSemantic.Assign)]
+		NSPersistentHistoryResultType ResultType { get; set; }
+
+		[NullAllowed, Export ("token", ArgumentSemantic.Strong)]
+		NSPersistentHistoryToken Token { get; }
+	}
+
+	[Watch (4,0), TV (11,0), Mac (10,13), iOS (11,0)]
+	[BaseType (typeof(NSPersistentStoreResult))]
+	interface NSPersistentHistoryResult
+	{
+		[Export ("result", ArgumentSemantic.Strong)]
+		NSObject Result { get; }
+
+		[Export ("resultType")]
+		NSPersistentHistoryResultType ResultType { get; }
+
+		[Export ("initWithResultType:andResult:")]
+		IntPtr Constructor (NSPersistentHistoryResultType type, NSObject result);
+
+		[Export ("initWithSubresults:")]
+		IntPtr Constructor (NSObject[] subresults);
+	}
+
+	[Watch (4,0), TV (11,0), Mac (10,13), iOS (11,0)]
+	[BaseType (typeof(NSObject))]
+	interface NSPersistentHistoryTransaction : NSCopying
+	{
+		[Export ("timestamp", ArgumentSemantic.Copy)]
+		NSDate Timestamp { get; }
+
+		[NullAllowed, Export ("changes", ArgumentSemantic.Copy)]
+		NSPersistentHistoryChange[] Changes { get; }
+
+		[Export ("transactionNumber")]
+		long TransactionNumber { get; }
+
+		[Export ("storeID")]
+		string StoreId { get; }
```

Comment:
Yeah @spouliot is right we use ID not as an acronym so it should be `Id` totally my bad